src/Data/Formatter/Number.purs

```diff
@@ -24,25 +23,39 @@ import Data.Either (Either, either)
 import Data.Int as Int
 import Data.String as Str
 
-import Data.Formatter.Internal (foldDigits, digit, repeat)
+import Data.Formatter.Parser.Utils (runP)
+import Data.Formatter.Internal (foldDigits, repeat)
+import Data.Formatter.Parser.Number (parseDigit)
 
 import Math as Math
 
 import Text.Parsing.Parser as P
 import Text.Parsing.Parser.Combinators as PC
 import Text.Parsing.Parser.String as PS
 
-type Formatter =
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Generic.Rep.Eq (genericEq)
+
+
+newtype Formatter = Formatter

```

Comment:
Here I'm just using pattern matching and constructor directly but for someone it might be useful
⚡️